### PR TITLE
Kuz 357 update architecture documentation

### PR DESCRIPTION
**Pull request by ballinette**
Update documentation about architecture overview and request scenarios as well as schemas, to make it consistent to latest code updates.

Some of this documentation will probably need some improvements to make it easier to read and understand (especially about pub/sub scenario, that is quite tricky), but this PR is a first step to make at least the documentation up-to-date with the code.

### CODE REVIEW

**Comment by scottinet** [comment]
Great!
:+1:

File: docs/request_scenarios/pubsub.md [hunk 1]
```diff
@@ -21,16 +21,20 @@ The following diagram shows how two different clients, a Websocket and a MQ one,
 
 ![pubsub_scenario_details1](../images/kuzzle_pubsub_scenario_details1.png)
 
-\#1a. The client application opens a Websocket connection and emit a "subscribe" event with some filters.
-For instance, to be notified about new messages posted to the collection "users", containing a field "hobby" equals to "computer":
+\#1. The client application opens a Websocket or a MQ connection and emit a "subscribe" event with some filters.
```
Comment (xbill82) [inline docs/request_scenarios/pubsub.md]: "emit" -> "emits"

File: docs/request_scenarios/pubsub.md [hunk 2]
```diff
@@ -63,13 +68,17 @@ Sample message:
...
 
-\#3. The ```Funnel Controller``` validates the message and transfer it to the ```Subscribe Controller```
+\#3. The ```Funnel Controller``` validates the message and transfer it to the ```Subscribe Controller```.
+
+\#4. The ```Subscribe Controller``` calls the ```HotelClerk``` internal component to create the subscription.
+
+\#5. The ```HotelClerk``` calls the ```DSL``` component to get a formated filter related to the subscription (see [DSL Readme for more details](../../lib/api/dsl/README.md)).
 
-\#4. The ```Subscribe Controller``` calculates the pub/sub room related to this filter, adds an internal reference to the client, and returns the Room ID.
+\#6. The ```HotelClerk``` create a channel related to the filters and give it back to the ```Subscribe Controller```.
```
Comment (xbill82) [inline docs/request_scenarios/pubsub.md]: "create" -> "creates"

File: docs/request_scenarios/pubsub.md [hunk 3]
```diff
@@ -63,13 +68,17 @@ Sample message:
...
 
-\#5. Callback functions are triggered to transmit the response message back to the ```Websocket Router```
+\#7. The channel is send back to the Websocket (or MQ) Router through the internal components.
```
Comment (xbill82) [inline docs/request_scenarios/pubsub.md]: "send" -> "sent"

File: docs/request_scenarios/pubsub.md [hunk 4]
```diff
@@ -77,57 +86,58 @@ Sample response content:
...
-  }
-}
-```
+\#3. The ```Notifier``` uses the ```Notification Cache``` engine to store the mapping content/rooms into cache.
+
+\#4. The ```Notifier``` calls the ```HotelClerk``` to get the channels related to the rooms.
+
+\#5. The ```Notifier``` broadcasts the message to each related channel to the Websocket (or MQ) router.
+
+\#6. Finally, the router send the message to the clients who subscribed to it.
```
Comment (xbill82) [inline docs/request_scenarios/pubsub.md]: "send" -> "sends"

File: docs/request_scenarios/pubsub.md [hunk 5]
```diff
@@ -77,57 +86,58 @@ Sample response content:
...
 
-\#5. The ```Write Controller``` calculates the rooms whose filters match this new message, and calls the ```Notifier``` core component.
+\#2. The ```Notifier``` calls the ```Notification Cache``` to get the rooms related to the content.
 
+\#3. The ```Notifier``` calls the ```HotelClerk``` to get the channels related to the rooms.
 
-\#6. The ```Notifier``` component notifies the subscribing clients, either directly through Websocket, and/or via the MQ Broker.
+\#4. The ```Notifier``` broadcasts the message to each related channel to the Websocket (or MQ) router.
 
+\#5. Finally, the router send the message to the clients who subscribed to it.
```
Comment (xbill82) [inline docs/request_scenarios/pubsub.md]: "send" -> "sends"

File: docs/request_scenarios/read_mq.md [hunk 6]
```diff
@@ -24,26 +23,30 @@ Sample STOMP request: retrieve the document ```739c26bc-7a09-469a-803d-623c4045b
...
+}
 
 ^@
 ```
 
-\#1c. The broker notifies the MQ Listener with the incoming message
-
-
-\#2. The MQListener handles the input message and forward it to the ```Funnel Controller```.
+\#2. The MQ plugin handles the input message and forward it to the ```Funnel Controller```.
```
Comment (xbill82) [inline docs/request_scenarios/read_mq.md]: "forward" -> "forwards"

File: docs/request_scenarios/write.md [hunk 7]
```diff
@@ -53,11 +51,13 @@ Sample message: (**note that _persist = true_**)
 }
 ```
 
-\#2b. The router then subscribes a listener to the task queue to notify the client about his request status
-
 \#3. The ```Funnel Controller``` validates the message and forward the request to the ```Write Controller```
 
-\#4. \#5. The Write Hook intercepts the request and sends it to the internal broker. (see [Hooks Readme](../../lib/hooks/README.md) for more details about hooks)
+\#4. The ```Write Controller``` triggers the ```Plugins Manager``` with a "data:create" event.<br/>
+The ```Plugins Manager``` calls all pipes and hooks configured by the active plugins (see [Plugin's documentation](../plugins.md)), and finally trigger the "add" event of the ```Write Hook```.<br/>
```
Comment (xbill82) [inline docs/request_scenarios/write.md]: "trigger" -> "triggers"

File: docs/request_scenarios/write.md [hunk 8]
```diff
@@ -73,38 +73,34 @@ Detailed workflow:
...
 
-\#12. The listener that the ```Router Controller``` registered in step 2b receive a notification from the write worker...
-
-\#13. ... and forward it back to the client.
-
-\#14. The ```Notifier``` core component is informed about the new content.
+\#12. The ```Worker Listener``` that the ```Write Controller``` registered in step 5a. receive a notification from the ```Internal Broker```...
 
-\#15. The ```Notifier``` component calculates the filtered rooms related to this content, and calls the Notification Cache service to store the content/roomId relationship
+\#13. ... and forward it back to the ```Router Controller```...
```
Comment (xbill82) [inline docs/request_scenarios/write.md]: "forward" -> "forwards"

File: docs/request_scenarios/pubsub.md [hunk 9]
```diff
@@ -77,57 +86,58 @@ Sample response content:
...
-
 ![pubsub_scenario_details2](../images/kuzzle_pubsub_scenario_details2.png)
 
-\#1. A client sends a new content to Kuzzle, either through a HTTP request, a Websocket connection or a MQ client (see [Reading scenarios](README.md#Reading-content-from-Kuzzle))
+\#1. A new content is published to the ```Notifier``` component.
 
-\#2. The router handles the input request and transmit the message to the ```Funnel Controller```.
+The "_publish_" method can be triggered:
+* either direclty by the ```Write Controller``` for non persistent data (using the [publish](http://kuzzleio.github.io/kuzzle-api-documentation/#publish) action).
+* or by by the ```Plugins Manager``` when a 'data:create' event is triggered, to notify users in real-time before the data are sent to the storage Engine.
```
Comment (dbengsch) [inline docs/request_scenarios/pubsub.md]: by by => by

**Comment by j33f** [comment]
corrections proposed by @xbill82 & @dbengsch applied

+1

**Comment by dbengsch** [comment]
+1

**Pull request merged**